src/rgw/rgw_rest_s3.h

```diff
@@ -119,6 +119,19 @@ class RGWListBucket_ObjStore_S3 : public RGWListBucket_ObjStore {
   void send_versioned_response();
 };
 
+class RGWListBucket_ObjStore_S3v2 : public RGWListBucket_ObjStore {
```

Comment:
RGWListBucket_ObjStore_S3V2 can inherit from RGWListBucket_ObjStore_S3 - this way a lot of code duplication can be avoided.